Analysis/PIDTools/src/LikelihoodPIDProcessor.cc

```diff
@@ -243,6 +258,25 @@ void LikelihoodPIDProcessor::init() {
   _mupiPID = new LowMomentumMuPiSeparationPID_BDTG(_weightFileName);
 
   printParameters();
+
+
+  //CHECK that init settings are correct
+  bool allnamecorrect=false;
+  for(int i=0; i<_methodstorun.size(); i++) {
+    {
+      if(  _methodstorun.at(i).compare("dEdxPID")==0 
+	   ||  _methodstorun.at(i).compare("LowMomMuID")==0 
+	   ||  _methodstorun.at(i).compare("BasicVariablePID")==0
+	   ||  _methodstorun.at(i).compare("ShowerShapesPID")==0
+	   ||  _methodstorun.at(i).compare("LikelihoodPID")==0 ) allnamecorrect=true;
+      
+      if(allnamecorrect==false) {
+	streamlog_out(ERROR) << " Check your method names.. "<<_methodstorun.at(i)<<"  is not in the list of valid methods: BasicVariablePID LowMomMuID ShowerShapesPID dEdxPID LikelihoodPID " <<std::endl;
+	raise(SIGSEGV);
```

Comment:
I just googled a solution to make my Marlin completely stop if this situation occurs. Probably not the more elegant solution...

Question: what happens when the exception is raised? Is Marlinstopped? Or does it shows the Exception message every event?

Reply:
In this case if you throw the exception Marlin will simply stop but display to the user the message of the exception before it does.

In general whenever, you simply want to stop Marlin because there is some condition that you cannot or do not want to handle throw an exception and do not catch it, Marlin will catch it, display the error message that is in the exception and stop the execution. However, if you throw an exception and catch it yourself to handle it, Marlin will not stop, unless you throw the exception again afterwards, so that Marlin can catch it in the end.

Reply:
So, I commit your sugestions.